### PR TITLE
chore: forward errors to sentry

### PR DESCRIPTION
**Pull request by devceline**
# Description

- add `logError` utility
- convert all `console.error` calls to `logError` (bar `firebase-messager` as we can't use sentry there)
# Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Draft PR (breaking/non-breaking change which needs more work for having a proper functionality _[Mark this PR as ready to review only when completely ready]_)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

# Fixes/Resolves (Optional)

Please list any issues that are fixed by this PR in the format `#<issue number>`. If it fixes multiple issues, please put each issue in a separate line. If this Pull Request does not fix any issues, please delete this section.

# Checklist:

- [x] I have performed a self-review of my own code
- [x] My changes generate no new warnings
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: tsconfig.json [hunk 1]
```diff
@@ -20,7 +20,7 @@
       "@/*": ["./src/*"]
     }
   },
-  "include": ["src", "playwright.config.ts", "tests"],
+  "include": ["src", "playwright.config.ts", "tests", "../../../../tmp/!home!clinenerd!dev!web3inbox!src!utils!error.ts~"],
```
Comment (bkrem) [inline tsconfig.json:23]: Intentional? What is this about if so?

File: src/utils/hooks/useNotifyProjects.ts [hunk 2]
```diff
@@ -1,10 +1,14 @@
 import { useContext, useEffect, useState } from 'react'
 
+import { captureException } from '@sentry/react'
```
Comment (bkrem) [inline src/utils/hooks/useNotifyProjects.ts:3]: Unused import
```suggestion
```

File: src/utils/projects.ts [hunk 3]
```diff
@@ -10,15 +12,14 @@ export async function fetchFeaturedProjects<T>() {
...
-      .then(async res => res.json())
-      .catch(err => console.error({ featuredProjects: err }))
+    const discoverProjectsData = await fetch(explorerUrlFeatured).then(async res => res.json())
     const discoverProjects = Object.values(discoverProjectsData.projects)
 
     return {
       data: discoverProjects as T
     }
   } catch (error) {
+    logError(error)
```
Comment (bkrem) [inline src/utils/projects.ts:22]: Not necessary here bc this is already a rethrow in the line below and we're seeing this in Sentry. Might cause dupes.


```suggestion
```

**Review by bkrem — APPROVED** [review]
LGTM 💯 minor corrections/cleanup

**Review by enesozturk — APPROVED** [review]

**Pull request merged**